### ISSUE TITLE
Support for GS-1331 (Kamvas 13)

### ISSUE THREAD

**Issue by CammiePone** [joseluis/huion-linux-drivers#82]
I've got the new Kamvas 13 (not the pro), and I'm really wanting to switch to Linux. Unfortunately, the drivers for my tablet and the art software are the only things I use preventing this switch.

I don't know how hard it would be to add the support, or if it would just work with one of the similar tablets in the config.ini, but I didn't see the Kamvas 13 in it, so I'm assuming it won't work.

If not, that's alright. I understand that it's probably a lot of work for a tablet not a lot of people are currently using.

**Comment by obook** [comment]
Hello CammiePone,
For Kamvas 13 may you test in config.ini
`current_tablet = [tablet_gt133]`
?
Olivier, Kamvas Pro 13 user under Ubuntu and Raspberry

**Comment by CammiePone** [comment]
I can try next chance I get, but I don't expect it to work perfectly. The Kamvas 13 uses a newer pen, has two buttons instead of a slider, and has some extra functionality that the Kamvas Pro 13 lacks.

**Comment by obook** [comment]
Hello CammiePone,
I think is the same pen. The slider is on a border of tablet for pro version.
The important thing is compile uclogic, install all requested softwares (xinput xserver-xorg-input-evdev python3-evdev python3-usb xdotool libnotify-bin arandr python3-numexpr) then create the xorg config file.
Good luck,
Olivier

**Comment by CammiePone** [comment]
The Kamvas Pro 13 uses the PW507,  whilst the Kamvas 13 uses the PW517. The touch bar of the pro was replaced by two extra buttons on the regular Kamvas 13, meaning I'd likely lose out on functionality there. While I'm sure I could get a hackjob working, I'd be missing pieces that I consider vital to drawing, hence why I made the issue to add support for the tablet.

**Comment by yoanndaly** [comment]
I've also got the new Kamvas 13 and I use Ubuntu 18.04. I've only tried a slight improvement to the solution suggested by Olivier.
I wrote the minimal variant of the configuration for Kamvas Pro 13 with corrected button information which goes as follows:

`[tablet_gs1331]`
`model_name   = GS1331`
`screen        = true`
`screen_width  = 1920`
`screen_height = 1080`
`pen_max_x     = 58752`
`pen_max_y     = 33048`
`pen_max_z     = 8191`
`resolution    = 5080`
`buttons       = 8`
`pen_min_tilt  = -60`
`pen_max_tilt  = 60`

Good news: If I define an 8 button menu, and execute the script with `debug_mode     = false ` the 8 buttons work just fine.

Also with hid-uclogic the pointer works well and is well calibrated even in the corners (but only in miror screen mode). So as soon as I unplug and plug the tablet, and manually turn to miror screen mode (with xrandr in terminal), the new pen PW517 works (but not the 8 buttons).
 
Bad news for me: 
When I execute the script, the pointer is disabled. When I  execute the script with `debug_mode     = true `, I suffer the bug [#41 ](https://github.com/joseluis/huion-linux-drivers/issues/41) (seems to me similar to [#65 ](https://github.com/joseluis/huion-linux-drivers/issues/65) )

So I think I haven't really tested the above calibration, only the configuration of 8 buttons. I will write more on one of the posts for the xinput bug.

Yoann

**Comment by yoanndaly** [comment]
I continue  my previous comment. I have finally solved my installation issue of my Kamvas 13, as explained in a comment of the closest bug I have found [#41 ](https://github.com/joseluis/huion-linux-drivers/issues/41). So I can confirm that if one follows the installation instructions for the driver, with the above configuration I suggested yesterday `current_tablet = [tablet_gs1331]`, then both the buttons and pen work fine and are properly calibrated (at least in miror display). Note that, with respect to the Kamvas Pro 13, I removed the line `scrollbar     = 1` since there is no scrollbar on the Kamvas 13. Note also that the calibration values are exactly those given by the debug mode (based on uclogic-decode), so that they should really be the correct values.

Good luck,
Yoann

**Comment by CammiePone** [comment]
Wait wait, so you got it perfectly function, Yoann? Nothing's missing?

**Comment by yoanndaly** [comment]
Yes, I think I got it perfectly function. The script works as it is for me. I have now even tried the multi-monitor settings. They work as long as one uses the native resolution of the Pen Display. I suggested in #87 a way to modify slightly the script to also allow other resolutions and rotations of the Tablet. One can also use the new PenTablet mode (without display). You can find there my config.ini with an example of my first button configuration, in case this may help. I have to say I didn't use Windows in more than 15 years and I don't intend to change that, so I can't really compare with what is provided there. This is my first Huion Tablet. I hope that you will switch to Linux soon and let me know if something is still missing according to you.  (Finally, note that I can't try the optional USB-C to USB-C cable since I don't have a USB-C port with DP-alt mode. So I don't know if this functionality works with the script).
